sdk/storage/src/data_lake/clients/file_system_client.rs

```diff
@@ -56,13 +55,11 @@ impl FileSystemClient {
         options: FileCreateOptions<'_>,
     ) -> Result<FileCreateResponse, crate::Error> {
         let mut request = self.prepare_file_create_request(file_path);
-        let contents = DataLakeContext {};
-        let mut pipeline_context = PipelineContext::new(ctx, contents);
 
         options.decorate_request(&mut request)?;
         let response = self
             .pipeline()
-            .send(&mut pipeline_context, &mut request)
+            .send(&ctx.create_override(), &mut request)
```

Comment:
This reads like creating an override without actually overriding anything.

Reply:
Good point. It's an unpleasant side effect of being unable to transform `Context` into a trait object 😢. If it were possible, `send` would accept it instead of a concrete `OverridableContext` instance.  

I'll try to come up with something better!

Reply:
@thovoll: what do you think now? I've replaced `create_override()` with `into()` when you do not want to override anything.

Reply:
Thanks @MindFlavor, looks better to me.